packages/search/src/ago/helpers/filters/encode-params.ts

```diff
@@ -0,0 +1,77 @@
+import { getProp } from "@esri/hub-common";
+
+/**
+ * Url-encoding of search params. This function is generic enough to encode a deeply nested object
+ * ```
+ * Example:
+ * Input: { a: { b: 2 }, c: 3 }
+ * Output: 'a[b]=2&c=3'
+ * ```
+ * @param {Any} params (query params from hub indexer)
+ * @returns {String}
+ */
+export function encodeParams(params: any = {}): string {
+  // get raw paths
+  const paths = getPaths(params);
+  const flatPaths = paths.filter(path => {
+    return typeof getProp(params, path.join(".")) !== "object";
+  });
+  const parts: string[] = [];
+  // for each nested path, we want to surround it with `[]`
+  // i.e. if a path is like ['a', 'b'], we want encoding as 'a[b]=2' given the input object { a: { b: 2 }, c: 3 }
+  flatPaths.forEach(path => {
+    let str = "";
+    for (let i = 0; i < path.length; i++) {
+      if (i === 0) {
+        str += path[i];
+      } else {
+        str += `[${path[i]}]`;
+      }
+    }
+    const right = encodeURIComponent(getProp(params, path.join(".")));
+    const left = encodeURIComponent(str);
+    if (right) {
```

Comment:
Can the value be `null` or `undefined`? Because

```
encodeURIComponent(null)
> "null"

encodeURIComponent(undefined)
> "undefined"
```

Also, it is better to move this to the beginning of the function.

Reply:
It will never be null or undefined because the path is calculated on params. So the path will be present in the params object. I will add an extra check like 
```
const right = encodeURIComponent(getProp(params, path.join(".")) || "");
```

Reply:
> Also, it is better to move this to the beginning of the function.

Sorry, I didn't understand that comment ^

Reply:
I mean having an earlier exit, but it doesn't really matter.